### PR TITLE
[MODULAR] Allows better interaction with the silly slots

### LINKED ISSUE #19298: Unable to interact with lewd slots even with exposed genitals
## Round ID: 
7638
(Though I don't believe it is necessary) 

## Reproduction:

Basically, lewd slots cannot be interacted with unless you are completely naked. Expose/Hide Genitals command doesn't affect whether or not you can interact with the slot. This seems like an oversight and it'd be nice to have it be fixed if someone wants to subtly play with the slots.

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

Fixes https://github.com/Skyrat-SS13/Skyrat-tg/issues/19298

Redacted?

## How This Contributes To The Skyrat Roleplay Experience

Fixes an oversight, lets you more subtly adjust things for better qol and RP potential.

## Proof of Testing

<details>
<summary>(NSFW) After exposing you may access the slot without removing clothing</summary>
  
dreamseeker_SkjhkISBB2

dreamseeker_EmomBC86HB

dreamseeker_aXsyB41wRr

</details>

## Changelog

:cl:
fix: you may now interact with lewd slots when the appropriate genitals are exposed (using the Expose/Hide genitals verb -> set to Always Show) regardless of whether or not you are clothed.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/interaction_menu/code/interaction_component.dm [hunk 1]
```diff
@@ -212,9 +212,19 @@
 		return FALSE
 	if(!slot_index) // This condition is for the UI to decide if the button is shown at all. Slot index should never be null otherwise.
 		return TRUE
-	if(slot_index == ORGAN_SLOT_NIPPLES && !target.is_topless())
-		return FALSE
-	return target.is_bottomless()
+	switch(slot_index)
+		if(ORGAN_SLOT_NIPPLES)
+			var/tits_out = target.has_breasts(required_state = REQUIRE_GENITAL_EXPOSED)
+			if(!tits_out)
```
Comment (softcerv) [inline modular_skyrat/modules/interaction_menu/code/interaction_component.dm]: you don't have to use a variable for this ((also you if you keep the variable, might want to use something more neutral for the variable name like `chest_exposed`))

File: modular_skyrat/modules/interaction_menu/code/interaction_component.dm [hunk 2]
```diff
@@ -212,9 +212,19 @@
...
+				return target.is_topless() // for when we don't have breasts
+			else
+				return TRUE
```
Comment (softcerv) [inline modular_skyrat/modules/interaction_menu/code/interaction_component.dm]:
```suggestion
				return TRUE
				
```

**Comment by YakumoChen** [comment]
Please use a github friendly PR title (and possibly body as well)

File: modular_skyrat/modules/interaction_menu/code/interaction_component.dm (hunk 1 above)
Comment (vinylspiders) [reply modular_skyrat/modules/interaction_menu/code/interaction_component.dm]: I couldn't help myself. edit: and the var did have a point, but I think in one of my revisions I accidentally removed it. Thanks for catching that

**Comment by vinylspiders** [comment]
> Please use a github friendly PR title (and possibly body as well)

Is this really a concern? Ok.

File: modular_skyrat/modules/interaction_menu/code/interaction_component.dm (hunk 1 above)
Comment (softcerv) [reply modular_skyrat/modules/interaction_menu/code/interaction_component.dm]: Very understandable, although we are trying to keep things clean on here.

**Review by lessthnthree — APPROVED** [review]

**Pull request merged**